gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -407,8 +409,10 @@ ChannelCredentials createMtlsChannelCredentials() throws IOException, GeneralSec
   }
 
   private ManagedChannel createSingleChannel() throws IOException {
+    Map<String, String> mergedHeaders = mergeHeadersWithCredentialHeaders();
```

Comment:
Off the top of my head I am a bit worried about how this may impact channel pooling. Our channel pools may constant recreate channels at some frequency and this may end up calling `getRequestMetadata()` over and over.

IIRC, some downstream teams have some latency requirements and this may end up slowing down applications.

Reply:
thanks for bringing this up.  I re-work this to be part of the build() process

Reply:
ok updated so headers are deduped as part of the transportprovider build process